### PR TITLE
Refactor backend and fix implicit login

### PR DESCRIPTION
**Pull request by lesterfernandez**
### Frontend
The `Response.json()` method return a promise, so we need to await it.

We don't need to return `null` from the loader. Returning nothing (which means `undefined` by default) is fine. 

Fun fact: due to some poor decisions made early in the creation of TypeScript, returning nothing has a type of `void` in TypeScript, even though the value is actually `undefined`. I updated the type of our loader data accordingly, except I used the `ReturnType` utility type from typescript. Now the type is inferred. 

Screenshot 2023-10-19 at 12 53 18 AM

### Backend
The docker compose file had some issues. It was declaring that the container should inherit some environment variables from the shell instance, since they were being declared but not set. However, they were then getting overwritten by the loading of a `env_file`. So, I removed the redundant environment variable declarations. Also, the health check was giving me some errors and saying that the variables it was attempting to access didn't exist, so we should revisit that later.

`gotdotenv.Load()` looks for a file named `.env` by default, so no need to pass in that as an argument.

Screenshot 2023-10-19 at 12 59 54 AM

The cors package's debug logger was spamming standard-out, making it hard to read important print statements while debugging.

I removed some old comments and refactored some code to simplify it. A good rule of thumb is that comments should explain why some code was written, not what it does. If the code itself uses good variable names (like ours currently does), then we don't need many comments since the code is readable already.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 func main() {
-	dotEnvErr := godotenv.Load(".env")
+	dotEnvErr := godotenv.Load()
```
Comment (lesterfernandez) [inline server/server.go:16]: `godotenv.Load()` looks for a  file named `.env` by default

**Review by JrodanDiaz — APPROVED** [review]
The backend looks good to me 👍

**Review by fnotorious — APPROVED** [review]
Looks good!

**Comment by lesterfernandez** [comment]
Thanks guys 🚀

**Pull request merged**